src/main/tut/simple-algebraic-data-types.md

```diff
@@ -0,0 +1,375 @@
+```Haskell
+swap :: (a, b) -> (b, a)
+swap (x, y) = (y, x)
+```
+```tut:silent
+def swap[A, B](x: A, y: B): (B, A) = (y, x)
+```
+................
+```Haskell
+((a, b), c)
+```
+```scala
+((A, B), C)
+```
+...............
+```Haskell
+(a, (b, c))
+```
+```scala
+(A, (B, C))
+```
+...............
+```Haskell
+alpha :: ((a, b), c) -> (a, (b, c))
+alpha ((x, y), z) = (x, (y, z))
+```
+```tut:silent
+def alpha[A, B, C]: ((A, B), C) => (A, (B, C)) =
+  { case ((x, y), z) => (x, (y, z)) }
```

Comment:
I went with a parameterless method here since all we really care about is destructuring the tuple, but I'm not sure how idiomatic these types of methods are outside of a class or object body.

Reply:
I think this choice is sensible.